scripts/deploy.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Script to deploy skynet-js to an hns domain.
+ *
+ * Example usage: SKYNET_JS_DEPLOY_SEED="..." node ./scripts/deploy.js
+ *
+ * You can generate the required seed with `genKeyPairAndSeed`.
+ *
+ * The first time you run this for a given hns domain, there won't be any data
+ * on the domain. Setting `skipDownload` will skip the download. After the
+ * upload, set the TXT record for the hns domain to the resulting resolver
+ * skylink.
+ */
+
+/* eslint-disable @typescript-eslint/no-var-requires */
+
+const {
+  genKeyPairFromSeed,
+  SkynetClient,
+  stringToUint8ArrayUtf8,
+  uriSkynetPrefix,
+} = require("@skynetlabs/skynet-nodejs");
+
+const fs = require("fs");
+const fse = require("fs-extra");
+const process = require("process");
+const tar = require("tar-fs");
+
+// The env var with the secret seed phrase to deploy with.
+const deploySeedEnvVar = "SKYNET_JS_DEPLOY_SEED";
+// The hns domain to deploy to.
+const hnsDomain = "skynet-js";
+// The location of the bundle to deploy. Must be a folder.
+const bundlePath = "bundle";
+// Location of package.json, used to get the latest version.
+const packageJson = "../package.json";
+// Set to true to skip the download.
+const skipDownload = false;
+// Set to true to skip the upload.
+const skipUpload = false;
+
+const dataKey = "skynet-js";
+const versionsDir = "versions";
+const versionsTarFile = `${versionsDir}.tar`;
+
+(async () => {
+  const client = new SkynetClient("https://siasky.net");
+
+  // Get the latest version from package.json.
+
+  const version = require(packageJson).version;
+  console.log(`Version: ${version}`);
+
+  // Download the existing version directory.
+
+  if (fs.existsSync(versionsDir)) {
+    fs.rmSync(versionsDir, { recursive: true });
+  }
+  if (fs.existsSync(versionsTarFile)) {
+    fs.rmSync(versionsTarFile);
+  }
+
+  if (!skipDownload) {
+    try {
+      console.log(`Downloading HNS domain '${hnsDomain}' -> '${versionsTarFile}'`);
+      await client.downloadFileHns(versionsTarFile, hnsDomain, { format: "tar" });
+      // Untar to versions dir.
+      console.log(`Untarring '${versionsTarFile}' -> '${versionsDir}'`);
+      const writer = tar.extract(versionsDir, {
+        // Make sure all existing subfiles are readable.
+        readable: true,
+      });
+      await new Promise((resolve, reject) => {
+        fs.createReadStream(versionsTarFile).pipe(writer);
+        writer.on("finish", resolve);
+        writer.on("error", reject);
+      });
+
+      // Delete tar file.
+      fs.unlinkSync(versionsTarFile);
+    } catch (error) {
+      // If there was any error, stop. The initial directory should be uploaded manually.
+      console.log(error);
+      return;
```

Comment:
> The initial directory should be uploaded manually.

Q: Is this something we should log to the console instead of having it as a comment inside the script?

Reply:
Good catch. I actually don't remember why I added this section, it doesn't seem to do anything. Seems related to first-time uploads, but the new `--first-time` option that I'm about to add should account for that. Removed this section.

Reply:
Actually, I can leave it and add a message suggesting the use of `--first-time`.